Connie update auth

kasperg opened the pull request:
The current implementation takes the username directly and uses that
for display and email. However when using Adgangsplatformen the name
will actually be the CPR number.

To avoid entering or displaying such information unnecessarily we make the following changes:

- Add support for specifying valid user names by pattern
- Only display the first six ciphers of CPR-like names
- Hash name for use in user email address.

Also a few other tweaks regarding Connie and Single Signon.

kasperg committed:
Scramble single signon names when using Connie

The current implementation takes the username directly and uses that
for display and email. However when using Adgangsplatformen the name
will actually be the CPR number.

We should not display such information unnecessarily so scramble it
using a CRC32 - a hash function which will generate a reasonable
short and probably unique value suitable for display.

kasperg committed:
Rework Connie single signon password checking

Instead of mucking around with values we might as well just check if
anything has even set.

kasperg committed:
Remove support for blocked single signon users in Connie

We cannot control the username so there is little reason to check
for certain string-based values.

kasperg committed:
Allow Connie to match on regular patterns

This way we can configure the provider to accept logins without
knowing the full name. This can contain a CPR number which we do not
want to expose to administrators.

kasperg committed:
Hide final four ciphers of displayed name if it looks like a CPR

This makes it easier to users to verify that they have logged in as
the intended user.

xendk commented on modules/ding_provider/connie/connie.auth.inc:
```diff
@@ -10,30 +10,29 @@
+
+  // We simply check that the name matches one of the predefined values or
+  // patterns, which is basically what we do at single sign on as log in is
+  // validated externally.
```
```php
// Allow users in connie_user_passes or who matches connie_auth_patterns, as there's no password involved in single signon.
```

Ref the post Arne posted: https://www.arp242.net/comments.html

xendk reviewed (approved):
:+1: just a comment change.

kasperg committed:
Implement separate OpenPlatform token provider

This module provides a dummy implementation of token handling for 
Open Platform without requiring the user to perform a single signon
with Adgangsplatformen.

kasperg committed:
Remove token debug override handling from Adgangsplatformen

It is no longer needed as you can use the Connie Openplatform token
module to manually set a token instead.

kasperg committed:
Make Open Platform token provider required

Instead of trying to jump through hoops to handle situations where
there is no Open Platform token provider we can instead just let 
users enable either Adgangsplatformen or the new Connie module.

kasperg committed:
Enable Connie Openplatform token module when running tests

Otherwise they will fail. We cannot expect the Adgangsplatformen
module to be enabled and working for this environment.

kasperg committed:
Remove token from browser when no longer available

This prevents it from lingering in the browser when it is no longer
available e.g. after the user has logged out.

xendk reviewed (approved):
:+1:
